### DISCUSSION TITLE
How can I filter all operations for specified user? Could tag feature resolve this problem?

### DISCUSSION THREAD

**Discussion by betwins** [apache/skywalking#8641] [Q&A] [answered]
Hi, 
skywalking do well for trace the log across micro-services.  But does it support trace the log across the operations(http api) from specified user.  For example, user james report a bug, I want to know what james operate(http api) recently to investigate the bug, how can I filter the api call by the user id? does using tag feature achieve the goal?

**Comment by wu-sheng** [comment, marked as answer]
If you could put user ID in the span tag, then OAL could filter accordingly.

**Reply by betwins** [reply]
> If you could put user ID in the span tag, then OAL could filter accordingly.

@wu-sheng  it seems that every function which needs to log with user id tag must be annotated by @TAG. as the following code shows
```
    @Trace(operationName = "customTraceFunction")
    @Tags({@Tag(key = "plaintext", value = "arg[0]"), @Tag(key = "ciphertext", value = "returnedObj"),@Tag(key = "biz.id", value = "arg[0]")})
    private String trace2(String plaintext) {

        log.info("traceId", TraceContext.traceId());
        return new Base64().encodeToString(plaintext.getBytes(StandardCharsets.UTF_8));
    }
```
Is there a basic method to log common tag with user id for all log line?

**Reply by wu-sheng** [reply]
Log or Tag? I am confused. This ID is in your private codes, there is nothing I can do. You could adjust your own codes or write private plugins, but the community has nothing to do, I am afraid.

**Reply by betwins** [reply]
Hi, @wu-sheng thanks a lot for your reply.
my needs can be understood as one of the following two expressions
1. I have the log line as below.  how can I filter the api call with log line including [UID: james] in skywalking-ui 
![image](https://user-images.githubusercontent.com/79825922/157423489-b629a6ee-1cd1-4441-a364-30c244df5c1f.png)
2.  How can I add tag uid for every api call? could it be achieved by using java agent or other method supported by skywalking?
![image](https://user-images.githubusercontent.com/79825922/157421756-ef0570c4-04dc-4d82-939f-99ec18e1cbc1.png)

**Reply by wu-sheng** [reply]
(1) Use LAL script to get ID and tag the logs.
(2) I think I have replied. Your private codes are not the thing I could know. I have given you 2 ways.

**Comment by betwins** [comment]
@wu-sheng  Thanks a lot! I have got it.
